packages/skeleton/src/lib/utilities/Modal/Modal.svelte

```diff
@@ -153,18 +161,22 @@
 	}
 
 	function onClose(): void {
-		if ($modalStore[0].response) $modalStore[0].response(false);
+		if (currentModal.response) currentModal.response(false);
 		modalStore.close();
 	}
 
 	function onConfirm(): void {
-		if ($modalStore[0].response) $modalStore[0].response(true);
+		if (currentModal.type !== 'confirm') return;
+
+		if (currentModal.response) currentModal.response(true);
```

Comment:
This is one isn't breaking change actually per se since this function is only called if modalType is "confirm" - unless we consider that it's type gets mutated somewhere between rerender and a callback call(impossible basically).

We could articulate this better if we introduce Modal sub-components to break-up the code a little bit - and this would eliminate the need to call this check.

Reply:
> This is one isn't breaking change actually per se since this function is only called if modalType is "confirm"

Sorry, I should've been more specific on this one. I meant that feel the same way as I did for the aside I wrote for using string literals instead of `true` or `false` for `onClose` and `onConfirm`. We would respond with `CLOSED` and `CONFIRMED`, respectively.